src/qt/clientmodel.cpp

```diff
@@ -80,19 +83,28 @@ void ClientModel::setMasternodeList(const CDeterministicMNList& mnList)
 
 CDeterministicMNList ClientModel::getMasternodeList() const
 {
-    LOCK(cs_mnlinst);
+    TRY_LOCK(cs_mnlinst,lock);
+    if (!lock)
+        return CDeterministicMNList();
+
     return mnListCached;
 }
 
 void ClientModel::refreshMasternodeList()
 {
-    LOCK(cs_mnlinst);
+    TRY_LOCK(cs_mnlinst,lock);
+    if(!lock){
+        return;
+    }
     setMasternodeList(deterministicMNManager->GetListAtChainTip());
 }
 
 int ClientModel::getNumBlocks() const
 {
-    LOCK(cs_main);
+    TRY_LOCK(cs_main,lock);
+    if (!lock) {
+        return -1;
```

Comment:
in code sometimes this snippet is used
```
setNumBlocks(model->getNumBlocks(), ... )
```
I'm pretty sure it'll cause crashes if `getNumBlocks()` can return -1